crates/ruma-client-api/src/rendezvous/create_rendezvous_session.rs

```diff
@@ -0,0 +1,72 @@
+//! `POST /_matrix/client/*/rendezvous/`
+//!
+//! Create a rendezvous session.
+
+pub mod unstable {
+    //! `msc4108` ([MSC])
+    //!
+    //! [MSC]: https://github.com/matrix-org/matrix-spec-proposals/pull/4108
+
+    use http::header::{CONTENT_TYPE, ETAG, EXPIRES, LAST_MODIFIED};
+    use ruma_common::{
+        api::{request, response, Metadata},
+        metadata,
+    };
+    use url::Url;
+
+    const METADATA: Metadata = metadata! {
+        method: POST,
+        rate_limited: true,
+        authentication: None,
+        history: {
+            unstable => "/_matrix/client/unstable/org.matrix.msc4108/rendezvous",
+        }
+    };
+
+    /// Request type for the `POST` `rendezvous` endpoint.
+    #[request]
+    pub struct Request {
+        /// Any data up to maximum size allowed by the server.
+        #[ruma_api(body)]
+        pub content: String,
+
+        /// The content type, must be of the `text/plain` type.
+        #[ruma_api(header = CONTENT_TYPE)]
+        pub content_type: String,
+    }
+
+    impl Request {
+        /// Creates a new `Request` with the given content.
+        pub fn new(content: String) -> Self {
+            Self { content, content_type: "text/plain".to_owned() }
+        }
+    }
+
+    impl Default for Request {
+        fn default() -> Self {
+            Self::new("".to_owned())
+        }
+    }
+
+    /// Response type for the `POST` `rendezvous` endpoint.
+    #[response]
+    pub struct Response {
+        /// The absolute URL of the rendezvous session.
+        pub url: Url,
+
+        /// ETag for the current payload at the rendezvous session as
+        /// per [RFC7232](https://httpwg.org/specs/rfc7232.html#header.etag).
+        #[ruma_api(header = ETAG)]
+        pub etag: String,
+
+        /// The expiry time of the rendezvous as per
+        /// [RFC7234](https://httpwg.org/specs/rfc7234.html#header.expires).
+        #[ruma_api(header = EXPIRES)]
+        pub expires: String,
```

Comment:
This should probably use a `web_time::SystemTime`, via the date_header crate, like we do in [`RetryAfter`](https://docs.ruma.dev/ruma_client_api/error/enum.RetryAfter.html).